File: solutions/typescript/darts/1/darts.ts
export function score(x: number, y: number): number {
  // Calculate the distance from the center (0, 0)
  const distance = Math.sqrt(x * x + y * y);
  
  // Check where the dart landed
  if (distance > 10) {
    return 0; // Outside the target
  } else if (distance > 5) {
    return 1; // Outer circle
  } else if (distance > 1) {
    return 5; // Middle circle
  } else {
    return 10; // Inner circle (bullseye)
  }
}

File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    int score(double xOfDart, double yOfDart) {
        // Calculate squared distance from center (0,0)
        double distanceSquared = xOfDart * xOfDart + yOfDart * yOfDart;
        
        if (distanceSquared <= 1.0) {
            return 10; // Inner circle (radius 1)
        } else if (distanceSquared <= 25.0) {
            return 5;  // Middle circle (radius 5)
        } else if (distanceSquared <= 100.0) {
            return 1;  // Outer circle (radius 10)
        }
        
        return 0; // Outside the target
    }
}